### PR TITLE
feat(#84): bookmarks and current distractions

### PR DESCRIPTION
**Pull request by lipefree**
Current distractions were not implemented because it was not clear enough what was supposed to be done and I didn't have time to spend more time on this task. I moved the UI part of distraction list to match branch 62 that will solve #62. 

You can now bookmarks distractions and you can filter them in the list and see which distraction you bookmarked.

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -38,6 +38,7 @@ android {
         sourceCompatibility JavaVersion.VERSION_1_8
         targetCompatibility JavaVersion.VERSION_1_8
     }
+
```
Comment (bluedrack) [inline app/build.gradle:41]: Not necessary

File: app/src/main/java/com/github/studydistractor/sdp/StudyDistractorScreen.kt [hunk 2]
```diff
@@ -61,7 +62,10 @@ fun StudyDistractorApp(
     )
     val context = LocalContext.current
     val distractionViewModel: DistractionViewModel = viewModel()
-    val distractionListViewModel = DistractionListViewModel(distractionService)
+//    val distractionListViewModel = DistractionListViewModel(distractionService)
```
Comment (bluedrack) [inline app/src/main/java/com/github/studydistractor/sdp/StudyDistractorScreen.kt:65]: You can remove the comment I think

File: app/src/main/java/com/github/studydistractor/sdp/bookmark/FirebaseBookmarks.kt [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+import com.google.firebase.database.ValueEventListener
+import javax.inject.Inject
+
+class FirebaseBookmarks @Inject constructor() : BookmarkModel {
+
+    private val auth = FirebaseAuth.getInstance()
+    private val db = FirebaseDatabase.getInstance()
+    private var bookmarks = SnapshotStateList<String>()
+    private val BOOKMARKSPATH = "Bookmarks"
+    private val databaseRef = db.getReference(BOOKMARKSPATH).child(auth.uid!!)
```
Comment (bluedrack) [inline app/src/main/java/com/github/studydistractor/sdp/bookmark/FirebaseBookmarks.kt:21]: auth.uid is null when in CI pay attention to it.

File: app/src/main/java/com/github/studydistractor/sdp/distraction/Distraction.kt [hunk 4]
```diff
@@ -16,7 +16,8 @@ data class Distraction(
     val long: Double? = null,
     val length : Length? = null,
     val tags: List<String>? = null,
-    val iconName: String? = null
+    val iconName: String? = null,
+    var distractionId: String? = null
```
Comment (bluedrack) [inline app/src/main/java/com/github/studydistractor/sdp/distraction/Distraction.kt:20]: I prefer the "uid" name for this variable it is in the same format as in uid for the user

File: app/src/main/java/com/github/studydistractor/sdp/distractionList/DistractionListModel.kt [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+
+import com.github.studydistractor.sdp.distraction.Distraction
+
+interface DistractionListModel {
+    /**
+     * Get all available distractions
+     */
+    fun getAllDistractions() : List<Distraction>
+
+    fun getTags() : List<String>
```
Comment (bluedrack) [inline app/src/main/java/com/github/studydistractor/sdp/distractionList/DistractionListModel.kt:11]: You can add a comment for this function

File: app/src/main/java/com/github/studydistractor/sdp/distractionList/DistractionListServiceFirebase.kt [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+    val distractions =  mutableStateListOf<Distraction>()
+    val tags = mutableStateListOf<String>()
+
+    init {
+        distractionDatabaseRef.addValueEventListener(object : ValueEventListener {
+            override fun onDataChange(snapshot: DataSnapshot) {
+                distractions.clear()
+                for(distraction in snapshot.children) {
+                    val distractionItem = distraction.getValue(Distraction::class.java)
+                    distractionItem!!.distractionId = distraction.key
```
Comment (bluedrack) [inline app/src/main/java/com/github/studydistractor/sdp/distractionList/DistractionListServiceFirebase.kt:27]: When the distractionItem is null I can cause crash.

File: app/src/main/java/com/github/studydistractor/sdp/ui/components/DistractionListFilterPanel.kt [hunk 7]
```diff
@@ -0,0 +1,183 @@
...
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.github.studydistractor.sdp.distraction.Distraction
+
+@Composable
+fun DistractionListFilterPanel(
```
Comment (bluedrack) [inline app/src/main/java/com/github/studydistractor/sdp/ui/components/DistractionListFilterPanel.kt:25]: Add comment there for the composable

File: app/src/main/java/com/github/studydistractor/sdp/ui/components/DistractionListLengthButton.kt [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+import androidx.compose.material3.ButtonDefaults
+import androidx.compose.material3.Text
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.unit.dp
+import androidx.compose.runtime.*
+
+@Composable
+fun DistractionListLengthButton(
```
Comment (bluedrack) [inline app/src/main/java/com/github/studydistractor/sdp/ui/components/DistractionListLengthButton.kt:17]: Add also comment

File: app/src/main/java/com/github/studydistractor/sdp/ui/components/ListedDistraction.kt [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.unit.dp
+import com.github.studydistractor.sdp.distraction.Distraction
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Favorite
+import androidx.compose.material3.Icon
+import androidx.compose.ui.graphics.Color
+
+@Composable
+fun ListedDistraction(
```
Comment (bluedrack) [inline app/src/main/java/com/github/studydistractor/sdp/ui/components/ListedDistraction.kt:25]: Add comment

**Review by bluedrack — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/github/studydistractor/sdp/distraction/Distraction.kt (hunk 4 above)
Comment (lipefree) [reply app/src/main/java/com/github/studydistractor/sdp/distraction/Distraction.kt:20]: Mhh I think it is still better to have explicit naming in dataclass, but we can put a format for naming data for next sprint.

**Comment by lipefree** [comment]
We had a problem with git and end of lines handling. There is a new PR for the same functionality.

**Pull request closed without merging**